User/jcirce/safety docs (#177)
CooperUnion/selfdrive: docs/safety.md

# Safety Policies & Procedures

The importance of this safety guide is to:

1. Ensure nobody gets harmed and reduce the risk of getting harmed

> 1. How to prevent easily avoidable mistakes

Reviewer: Should read:
To prevent easily avoidable mistakes

> 1. Reduce damage

Reviewer: Reduce damage to components/equipment

## General

> 

Reviewer: ADD:
All members should know where E-STOP buttons are.

*If something starts smoking:*

- Press the E-STOP and turn off all power immediately
- Clear the lab
- Call for help

*In case of fire:*

- There is a fire extinguisher on inside of the g-wagon
- There is a larger fire extinguisher right outside 701

*If something gets in the eye:*

- There is an eye wash in each 701 sink
- Do not rub eye
- Call for help

*If bleeding:*

- Call for help
- Apply pressure to the wound
- First-aid kit is near the exit of 701
- Wash hands
- Use disinfectant
- Use gauze/band-aids

*If bleeding doesn't stop:*

- Keep applying pressure

> - Go to urgent care

Reviewer: An accident report will have to be filled out. Inform technician/instructor of wound

  jack stands as needed)

*When operating power tools:*

- Make sure all is clear before using the tool
- Announce "ALL CLEAR, OPERATING (TOOL)" so that all know to be clear
- Ensure items are properly clamped before drilling so they don't go
  flying or break
- Use step bits when possible for drilling into plastic

*When wiring a battery:*

- Make sure all is clear before removing insulation to prevent shorts
- For anything loose use electric tape to prevent any accidental shorts
- Use battery boots (where possible)
- If a battery shorts, **stop** what you're doing, turn off all systems
  to prevent further damage, and identify the cause for the short

*When soldering:*

- Ensure you have appropriate ventilation
- Ensure soldering iron is not left on when finished

*When using a heat gun:*

- Only point at items you want to heat
- Keep clear of people and other objects when the heat gun is on
- Ensure the heat gun is turned off when finished

## Test Day Procedures

*In general:*

- If not directly testing, **stay clear**
- Do not distract people testing as it may cause a lapse in judgement
- Save non-urgent questions for when people are not actively testing

> 

Reviewer: When arming vehicle, announce "ARMING VEHICLE"
Same when disarming vehicle.
When test is starting, announce "STARTING TEST"

*Test day safety officer:*

- One person is assigned to be safety officer for the day
- Ensures people follow safety protocols
- Designated as person to call 911 and use the fire extinguisher
- Responsible for asking people not following safety protocol to leave
  the test environment

*When a car is armed:*

- A car is considered armed if the key is in the ignition or
  microcontrollers are powered

> - Nobody is to be in front of an armed car, **no exceptions**

Reviewer: Nobody is to be in FRONT OR BEHIND the car when armed, **no exceptions**

> # Autonomy Lab Safety Policies & Procedures

Reviewer 2: ```suggestion
# Safety Policies & Procedures
```

It's redundant to put `Autonomy Lab` in the title.

Document serves purpose of safety guidelines (both in 701 and outside)

### The importance of this safety guide is to ensure:

1. **NO ONE GETS HARMED** or we mitigate the risk of people getting harm
1. There isn’t a lot of capital damage to the car or anything in the
   surrounding area
1. Helps prevents any easily avoidable mistakes that could be prevented

>    and set back the timeline

Reviewer 2: ```suggestion
The importance of this safety guide is to:

1. Ensure nobody gets harmed and reduce the risk of getting harmed
2. How to prevent easily avoidable mistakes
3. Reduce damage
```

## General

### If something starts smoking:

- Press estop and turn off all power immediately
- Clear the lab asap

> - Call for help

Reviewer 2: ```suggestion
*If something starts smoking:*

- Press the E-STOP and turn off all power immediately
- Clear the lab
- Call for help
```

### If fire:

- Fire extinguisher is on inside of car

> - Larger one in front of 701

Reviewer 2: ```suggestion
*In case of fire:*

- There is a fire extinguisher on inside of the g-wagon
- There is a larger fire extinguisher right outside 701
```

### If something in eye:

- Eye wash is on either sink in 701
- Do not rub eye

> - Call for help

Reviewer 2: ```suggestion
*If something gets in the eye:*

- There is an eye wash in each 701 sink
- Do not rub eye
- Call for help
```

### If bleeding:

- Call for help
- Apply pressure to wound
- First Aid kit near exit of 701
- Use gauze/band aids
- Wash hands

> - Use disinfectant

Reviewer 2: ```suggestion
*If bleeding:*

- Call for help
- Apply pressure to the wound
- First-aid kit is near the exit of 701
- Wash hands
- Use disinfectant
- Use gauze/band-aids
```

### If bleeding is not stopping:

- Keep applying pressure
- Call for help

> - Go to urgent care

Reviewer 2: ```suggestion
*If bleeding doesn't stop:*

- Keep applying pressure
- Go to urgent care
```

You've already called for help at this point so it's redundant to say.

### If someone passes out / loses consciousness:

> - Call 911 NO EXCEPTIONS

Reviewer 2: ```suggestion
*If someone loses consciousness:*

- Call 911, **NO EXCEPTIONS**
```

## When Using Equipment:

### MUST:

- Wear goggles
- Wear electrical gloves (if handling +48V line)
- Place chock blocks on wheels if tires are not on dollies

This includes but not limited to: mechanical maintenance, using hand

> tools, using power tools, cutting/stripping wire, soldering

Reviewer 2: ```suggestion
## Using Equipment

You **must** wear goggles (and electric gloves if handling high
voltage). This includes, but is not limited to: mechanical maintenance,
using hand/power tools, cutting/stripping wire, soldering, etc...
```

Mentioning the wheel dollies feels out of place here.

### When jacking car:

- Announce “ALL CLEAR, JACKING FRONT” or “JACKING BACK”
- Announce “ALL CLEAR, LOWERING” when releasing the jack

> - Wait until clear before jacking or releasing jack

Reviewer 2: ```suggestion
*When jacking a car:*

- Wait until clear before jacking or releasing jack
- Announce "ALL CLEAR, JACKING (FRONT OR BACK)"
- Announce "ALL CLEAR, LOWERING" when releasing the jack
```

Be careful when pasting from Google Docs, it uses unicode double quotes as opposed to ASCII double quotes.

### When going under car:

- Announce / shout out “GOING UNDER CAR” to make sure everyone near the
  car knows not to sit inside the car or move the car
- Do not go under car until sure no one is going to sit in car / move
  car
- Put up caution tape or assign someone else to watch car or stand at
  door so people coming in know not to go into car while you are under

>   the car

Reviewer 2: ```suggestion
*When going under a car:*

- Put up caution tape or assign someone else to watch the car so people
  coming nearby know to stay clear of the car
- Do not go under the car until you are sure it is safe to do so
- Announce "GOING UNDER CAR" to make sure everyone near the car knows
  not to sit inside or move the car
```

Thing that I've noticed that seems to repeat throughout is some of these points are redundant or out of logical order :p

### When operating experimental boards / devices / new additions to car:

- Check that it works on a node not plugged into car first
- Check board is not being back-powered through battery device

> - Operate with power supply before connecting to electronics to car

Reviewer 2: ```suggestion
*When operating experimental components:*

- Confirm the component works on a lab bench before testing on a project
- Ensure there is a safe way to disengage the component
```

Author: i really like the safe way to disengage

### When car is in 701:

- Ensure car will not drive into wall (put car on dollies or use chock
  blocks at all times)
- Have the car face at the wall where it won’t hit the hallway or anyone
  else in case it drives forward by itself
- Avoid keeping the car on it’s wheels unless needed for testing

> - Even this should be avoided by using the jack stands in the lab

Reviewer 2: ```suggestion
*When there's a car in 701:*

- Ensure the car will not drive into a wall by putting the car on
  dollies, jack stands, or using chock blocks at all times
- Have the car face at a wall where it won't drive into the hallway or
  anyone else in case of unintentional movement
- Avoid keeping the car on it's wheels unless needed for testing
```

### When using power tools / drills:

- Announce/ shout out “ALL CLEAR, DRILLING” so that all in car know to
  be clear
- Make sure all are clear before using tool
- Ensure item you are drilling into is clamped in some way so that it
  doesn’t go flying or break

> - Use step bits when possible for drilling into plastic

Reviewer 2: ```suggestion
*When operating power tools:*

- Make sure all is clear before using the tool
- Announce "ALL CLEAR, OPERATING (TOOL)" so that all know to be clear
- Ensure items are properly clamped before drilling so they don't go
  flying or break
- Use step bits when possible for drilling into plastic
```

### When wiring to a battery (if battery leads are exposed):

- Make sure all are clear before removing tape / no one is touching
  other side of battery to prevent shorts
- If anything is loose or hanging use electric tape to prevent any
  accidental shorts
- Use battery boots (when possible)
- If battery shorts, STOP what you're doing, turn off all systems to

>   prevent further damage, identify problem.

Reviewer 2: ```suggestion
*When wiring a battery:*

- Make sure all is clear before removing insulation to prevent shorts
- For anything loose use electric tape to prevent any accidental shorts
- Use battery boots (where possible)
- If a battery shorts, **stop** what you're doing, turn off all systems
  to prevent further damage, and identify the cause for the short
```

### When soldering:

- Ensure some sort of fan/filter is on (either the ceiling air sucker or
  the mini fan)

> - Ensure soldering iron is not left on when finished

Reviewer 2: ```suggestion
*When soldering:*

- Ensure you have appropriate ventilation
- Ensure soldering iron is not left on when finished
```

### When using Heat-Gun

- Make sure ONLY pointing at items want to heat up
- Keep others clear when heat gun is on

> - Ensure heat gun is turned off when finished

Reviewer 2: ```suggestion
*When using a heat gun:*

- Only point at items you want to heat
- Keep clear of people and other objects when the heat gun is on
- Ensure the heat gun is turned off when finished
```

### When running new steel cables for Brake-by-Cable:

- Use swage tool in vice when possible
- Clean all steel shavings when cutting cables
- Tape/cover all sharp ends of the cable

> 

Reviewer 2: ```suggestion
```

This should be a part of Brake-By-Cable's documentation, maybe add a section on looking at associated component/project safety pages.

## Test Day Procedures

### Test day safety officer

- One person (can be volunteer) is assigned to be safety officer for day
- This person will make sure people are following the ALL CLEAR shouts
- Person makes sure no one is in car when people are under car
- Person makes sure people using tools are wearing goggles
- Will be the person to call 911 if needed
- Will be the first person to use the fire extinguisher if needed
- Responsible for asking people not being safe to leave the room/test
  environment

### In General

- If not directly touching / testing something on car, STAY CLEAR
- If have curiosity questions, it’s okay to ask but wait until the team
  member who is working is finished or save questions for reflections or
  other times
- If afraid you might forget your question, write it down somewhere
- Ideally, unless it’s urgent, don’t distract people using tools and
  working on car
- Not only is it important for team members to focus while using tools
  to work on the car, distractions can cause mistakes which could cause

>   the car to become unsafe

Reviewer 2: ```suggestion
## Test Day Procedures

*In general:*

- If not directly testing, **stay clear**
- Do not distract people testing as it may cause a lapse in judgement
- Save non-urgent questions for when people are not actively testing

*Test day safety officer:*

- One person is assigned to be safety officer for the day
- Ensures people follow safety protocols
- Designated as person to call 911 and use the fire extinguisher
- Responsible for asking people not following safety protocol to leave
  the test environment
```

### When car is ARMED

- Armed means any of the following: Key is in ignition, Node boards are
  powered
- Take extreme care not to short connections
- If possible, turn off car / nodes before using metal tools on the
  boards (metal snips, screw drivers, etc)
- If anything shorts / makes smoke/ explodes, STOP what you're doing,
  call for help, turn all systems OFF to prevent further damages,
  identify problem before moving on
- If working with something connected to a battery, take even more care

>   not to short components

Reviewer 2: ```suggestion
*When a car is armed:*

- A car is considered armed if the key is in the ignition or
  microcontrollers are powered
- Nobody is to be in front of an armed car, **no exceptions**
- A fire extinguisher must be inside the car at all times
- Avoid using tools
- Take extreme care not to short anything
- If anything shorts, smokes, or explores, **stop** immediately, turn
  off **all** systems, and call for help before identifying the problem
```

> ### When outside:
> 
> - Yellow caution tape and orange cones for to corral/block pedestrians
> - No one in front or behind car while car is “ARMED” NO EXCEPTIONS
> - Must have fire extinguisher in car

Reviewer 2: ```suggestion
*When outside of 41CS:*

- Secure the test area with caution tape and orange cones
- Have a megaphone in case an announcement needs to be made
```

> ### Before commencing Test Day run through safety checklist:
> 
> - Who is designated student safety officer?
> - Does the ESTOP work as intended?
> - Is the weather suitable for a test day?
> - Is there a lab technician or someone certified who agreed they’d be
>   with you during the duration of the test day?
> - Is the school grounds team aware of what section of the sidewalk will
>   be blocked for testing?
> - Make sure all members present for test day understand what system is
>   being tested when and are conscious of any dangers that maybe
>   associated with it

Reviewer 2: ```suggestion
*Before a test day, confirm the following:*

- Is there a designated safety officer?
- Are E-STOPs functional?
- Is weather suitable?
- Is there a lab technician or someone certified who agreed they'd be
  with you during the duration of the test?
- Is the school grounds team aware of what section of the sidewalk will
  be blocked off for testing?
- Ensure everyone participating is informed of the tests and the risks
  associated with them
```
